jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecr/actions/PushToRepositoryAction.kt

```diff
@@ -0,0 +1,365 @@
+
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.getRequiredData(LangDataKeys.PROJECT)
+        val client: EcrClient = project.awsClient()
+
+        val selectedRepository = e.getData(ExplorerDataKeys.SELECTED_NODES)
+            ?.mapNotNull { it as? EcrRepositoryNode }
+            ?.takeIf { it.size == 1 }
+            ?.first()
+            ?.repository
+
+        val dialog = PushToEcrDialog(project, selectedRepository, dockerServerRuntime)
+        val result = dialog.showAndGet()
+
+        if (!result) {
+            // user cancelled; noop
+            EcrTelemetry.deployImage(project, Result.Cancelled, project.activeRegion().id)
+            return
+        }
+
+        launch {
+            val pushRequest = dialog.getPushRequest()
+            var result = Result.Failed
+            try {
+                val authData = withContext(Dispatchers.IO) {
+                    client.authorizationToken.authorizationData().first()
+                }
+
+                val ecrLogin = authData.getDockerLogin()
+                EcrUtils.pushImage(project, ecrLogin, pushRequest)
+                result = Result.Succeeded
+            } catch (e: SdkException) {
+                val message = message("ecr.push.credential_fetch_failed")
+
+                LOG.error(e) { message }
+                notifyError(message("ecr.push.title"), message)
+            } catch (e: Exception) {
+                val message = message("ecr.push.unknown_exception")
+
+                LOG.error(e) { message }
+                notifyError(message("ecr.push.title"), message)
+            } finally {
+                val type = when (pushRequest) {
+                    is ImageEcrPushRequest -> EcrDeploySource.Tag
+                    is DockerfileEcrPushRequest -> EcrDeploySource.Dockerfile
+                }
+                EcrTelemetry.deployImage(
+                    project,
+                    result,
+                    regionId = project.activeRegion().id,
+                    ecrDeploySource = type
+                )
+            }
+        }
+    }
+
+    companion object {
+        private val LOG = getLogger<PushToRepositoryAction>()
+    }
+}
+
+internal data class LocalImage(
+    val imageId: String,
+    val tag: String?
+)
+
+internal class PushToEcrDialog(
+    private val project: Project,
+    selectedRepository: Repository?,
+    private val dockerServerRuntime: Deferred<DockerServerRuntimeInstance>
+) : DialogWrapper(project, null, false, IdeModalityType.PROJECT),
+    CoroutineScope by ApplicationThreadPoolScope("PushRepositoryDialog") {
+    private val defaultTag = "latest"
+    private var type: BuildType = BuildType.LocalImage
+    private var remoteTag: String = ""
+    private val localImageRepoTags = CollectionComboBoxModel<LocalImage>()
+
+    private var localImage: LocalImage? = null
+    private var runConfiguration: DockerRunConfiguration? = null
```

Comment:
Can we be consistent on specifying type vs not?

Reply:
fixed